File: wheystore/wheystore.Web/Modules/SanPhamDB/QuanHuyen/QuanHuyenColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.QuanHuyen")]
[BasedOnRow(typeof(QuanHuyenRow), CheckNames = true)]
public class QuanHuyenColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string TenQuanHuyen { get; set; }
    public string TinhThanhTenTinhThanh { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/QuanHuyenColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { QuanHuyenRow } from "./QuanHuyenRow";

export interface QuanHuyenColumns {
    Id: Column<QuanHuyenRow>;
    TenQuanHuyen: Column<QuanHuyenRow>;
    TinhThanhTenTinhThanh: Column<QuanHuyenRow>;
}

export class QuanHuyenColumns extends ColumnsBase<QuanHuyenRow> {
    static readonly columnsKey = 'SanPhamDB.QuanHuyen';
    static readonly Fields = fieldsProxy<QuanHuyenColumns>();
}